cpp/libcudf_kafka/src/kafka_consumer.cpp

```diff
@@ -153,6 +153,26 @@ int64_t kafka_consumer::get_committed_offset(std::string const &topic, int parti
   return offset > 0 ? offset : 0;
 }
 
+const std::map<std::string, std::vector<int32_t>> kafka_consumer::list_topics()
+{
+  RdKafka::Metadata *md;
+  std::map<std::string, std::vector<int32_t>> topic_parts;
+
+  CUDF_EXPECTS(RdKafka::ERR_NO_ERROR == consumer->metadata(true, nullptr, &md, default_timeout),
+               "Failed to list_topics in Kafka broker");
+
+  const RdKafka::Metadata::TopicMetadataVector *mdv = md->topics();
+
+  for (auto &topic : *mdv) {
```

Comment:
I think we don't really need the `mdv` variable, it's only used in the loop.
```suggestion
  for (auto const &topic : *std::unique_ptr<RdKafka::Metadata::TopicMetadataVector>{md->topics()}) {
```

Reply:
@vuule I like this but it doesn't compile ... ideas?

```/home/jdyer/Development/cudf/cpp/libcudf_kafka/src/kafka_consumer.cpp:167:78: error: no matching function for call to ‘std::unique_ptr<std::vector<const RdKafka::TopicMetadata*> >::unique_ptr(<brace-enclosed initializer list>)’
        *(std::unique_ptr<RdKafka::Metadata::TopicMetadataVector>{md->topics()})) {```